pkg/command/service/generate.go

```diff
@@ -0,0 +1,200 @@
+				}
+			} else {
+				return fmt.Errorf("both ns and nsPrefix are empty")
+			}
+			if checkReady {
+				generator.NewBatchGenerator(time.Duration(interval)*time.Second, count, batch, concurrency, nsNameList, createKSVC, checkServiceStatusReady).Generate()
+			} else {
+				generator.NewBatchGenerator(time.Duration(interval)*time.Second, count, batch, concurrency, nsNameList, createKSVC, func(ns, name string) error { return nil }).Generate()
+			}
+
+			return nil
+		},
+	}
+	// count, interval, batch, minScale, maxScale int
+	ksvcGenCommand.Flags().IntVarP(&count, "number", "n", 0, "Total number of ksvc to be created")
+	ksvcGenCommand.MarkFlagRequired("count")
+	ksvcGenCommand.Flags().IntVarP(&interval, "interval", "i", 0, "Interval for each batch generation")
+	ksvcGenCommand.MarkFlagRequired("interval")
+	ksvcGenCommand.Flags().IntVarP(&batch, "batch", "b", 0, "Number of ksvc each time to be created")
+	ksvcGenCommand.MarkFlagRequired("batch")
+	ksvcGenCommand.Flags().IntVarP(&concurrency, "concurrency", "c", 10, "Number of multiple ksvcs to make at a time")
+	// ksvcGenCommand.MarkFlagRequired("concurrency")
+	ksvcGenCommand.Flags().IntVarP(&minScale, "minScale", "", 0, "For autoscaling.knative.dev/minScale")
+	ksvcGenCommand.MarkFlagRequired("minScale")
+	ksvcGenCommand.Flags().IntVarP(&maxScale, "maxScale", "", 0, "For autoscaling.knative.dev/minScale")
+	ksvcGenCommand.MarkFlagRequired("minScale")
+
+	ksvcGenCommand.Flags().StringVarP(&nsPrefix, "nsPrefix", "p", "", "Namespace prefix. The ksvc will be created in the namespaces with the prefix")
+	ksvcGenCommand.Flags().StringVarP(&nsRange, "nsRange", "", "", "")
+	ksvcGenCommand.Flags().StringVarP(&ns, "ns", "", "", "Namespace name. The ksvc will be created in the namespace")
+
+	ksvcGenCommand.Flags().StringVarP(&svcPrefix, "svcPrefix", "", "testksvc", "ksvc name prefix. The ksvcs will be svcPrefix1,svcPrefix2,svcPrefix3......")
+	ksvcGenCommand.Flags().BoolVarP(&checkReady, "wait", "", false, "whether wait the previous ksvc to be ready")
+	ksvcGenCommand.Flags().DurationVarP(&timeout, "timeout", "", 10*time.Minute, "duration to wait for previous ksvc to be ready")
+
+	return ksvcGenCommand
+}
+
+func createKSVC(ns string, index int) (string, string) {
+	service := servingv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%d", svcPrefix, index),
+			Namespace: ns,
+		},
+	}
+
+	service.Spec.Template = servingv1.RevisionTemplateSpec{
+		Spec: servingv1.RevisionSpec{},
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				"autoscaling.knative.dev/minScale": fmt.Sprintf("%d", minScale),
+				"autoscaling.knative.dev/maxScale": fmt.Sprintf("%d", maxScale),
+			},
+		},
+	}
+	service.Spec.Template.Spec.Containers = []corev1.Container{
+		{
+			Image: "docker.io/qibobo/go4autoscaler:latest",
+			Ports: []corev1.ContainerPort{
+				{
+					ContainerPort: 8080,
+				},
+			},
+		},
+	}
```

Comment:
Those should be kind of joined together.
Also we should not check in this using some random image.

Reply:
How about switch to `gcr.io/knative-samples/knative-route-demo` from knative community doc?

Reply:
For now I updated to "gcr.io/knative-samples/helloworld-go".